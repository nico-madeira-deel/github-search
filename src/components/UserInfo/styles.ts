import styled, { css } from 'styled-components'

export const Card = styled.section`
  width: 100%;

  ${({ theme }) => css`
    background-color: ${theme.colors.white};
    color: ${theme.colors.darkBlue};
    border-radius: ${theme.border.radius};
    margin: ${theme.spacings.medium} 0;
    padding: ${theme.spacings.small};

    h2 {
      margin-bottom: ${theme.spacings.xxsmall};
    }

    p {
      font-size: ${theme.font.sizes.small};
    }
  `}
`

export const Wrapper = styled.div`
  align-items: center;
  display: flex;
  flex-direction: column;
  margin: 0 auto;
  max-width: 300px;
  text-align: center;
`

export const Picture = styled.picture`
  border-radius: 100%;
  display: block;
  height: 12rem;
  overflow: hidden;
  width: 12rem;

  img {
    height: auto;
    max-width: 100%;
  }

  ${({ theme }) => css`
    border: 4px solid ${theme.colors.darkBlue};
    margin-bottom: ${theme.spacings.xsmall};
  `}
`

export const Followers = styled.div`
  display: flex;

  ${({ theme }) => css`
    margin-top: ${theme.spacings.xsmall};

    a {
      background-color: ${theme.colors.pink};
      border-radius: ${theme.border.radius};
      color: ${theme.colors.darkBlue};
      font-size: ${theme.font.sizes.xsmall};
      font-weight: ${theme.font.bold};
      padding: ${theme.spacings.xxsmall};

      &:first-child {
        margin-right: ${theme.spacings.xxsmall};
      }
    }
  `}
`
