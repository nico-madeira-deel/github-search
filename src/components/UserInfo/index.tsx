import { useEffect, useState } from 'react'
import * as S from './styles'
import Link from 'next/link'

import { UserResponse } from 'models/user'

const UserInfo = ({
  avatar_url,
  bio,
  email,
  followers,
  following,
  login,
  name
}: UserResponse) => {
  const [userUrl, setUserUrl] = useState<string>('')

  useEffect(() => {
    if (login) {
      setUserUrl(`https://github.com/${login}?tab=`)
    }
  }, [login])

  return (
    <S.Card>
      <S.Wrapper>
        {avatar_url ? (
          <S.Picture>
            <img alt={`Avatar image for the user ${login}`} src={avatar_url} />
          </S.Picture>
        ) : null}
        <h2>{name || login}</h2>
        {bio ? <p>{bio}</p> : null}
        {email ? <p>{email}</p> : null}
        {userUrl ? (
          <S.Followers>
            <Link href={`${userUrl}followers`} passHref={true}>
              <a target="_blank" rel="noopener noreferrer">
                <small>Followers: {followers}</small>
              </a>
            </Link>
            <Link href={`${userUrl}following`} passHref={true}>
              <a target="_blank" rel="noopener noreferrer">
                <small>Following: {following}</small>
              </a>
            </Link>
          </S.Followers>
        ) : null}
      </S.Wrapper>
    </S.Card>
  )
}

export default UserInfo
