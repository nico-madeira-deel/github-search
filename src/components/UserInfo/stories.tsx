import { Story, Meta } from '@storybook/react/types-6-0'
import UserInfo from '.'
import { UserResponse } from 'models/user'
import { mockUser } from 'utils/tests/mock'

export default {
  title: 'UserInfo',
  component: UserInfo
} as Meta

export const Basic: Story<UserResponse> = (args) => (
  <UserInfo {...mockUser} {...args} />
)

Basic.args = mockUser
