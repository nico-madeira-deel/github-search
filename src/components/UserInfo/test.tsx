/* eslint-disable @typescript-eslint/no-unused-vars */
import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'
import { mockUser } from 'utils/tests/mock'

import UserInfo from '.'

describe('<UserInfo />', () => {
  it('should render the user info', () => {
    const { container } = renderWithTheme(<UserInfo {...mockUser} />)

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should have user image if prop avatar_url is passed', () => {
    renderWithTheme(<UserInfo {...mockUser} />)

    expect(
      screen.getByRole('img', {
        name: `Avatar image for the user ${mockUser.login}`
      })
    ).toBeInTheDocument()
  })

  it('should display heading with user name if prop name its passed', () => {
    renderWithTheme(<UserInfo {...mockUser} />)

    expect(
      screen.getByRole('heading', {
        name: mockUser.name
      })
    ).toBeInTheDocument()
  })

  it('should display heading with user login if prop name itsnt passed', () => {
    const { name, ...propsWithoutName } = mockUser
    renderWithTheme(<UserInfo {...propsWithoutName} />)

    expect(
      screen.getByRole('heading', {
        name: mockUser.login
      })
    ).toBeInTheDocument()
  })

  it('should display user email if prop email its passed', () => {
    renderWithTheme(<UserInfo {...mockUser} />)

    expect(screen.getByText(mockUser.email)).toBeInTheDocument()
  })

  it('should display user bio if prop bio its passed', () => {
    renderWithTheme(<UserInfo {...mockUser} />)

    expect(screen.getByText(mockUser.bio)).toBeInTheDocument()
  })
})
