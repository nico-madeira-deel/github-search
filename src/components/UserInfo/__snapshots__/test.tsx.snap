// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserInfo /> should render the user info 1`] = `
.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 0 auto;
  max-width: 300px;
  text-align: center;
}

.c1 h2 {
  margin-bottom: 0.8rem;
}

.c1 p {
  font-size: 1.4rem;
}

.c2 {
  border-radius: 100%;
  display: block;
  height: 12rem;
  overflow: hidden;
  width: 12rem;
  border: 4px solid #555b6e;
  margin-bottom: 1.6rem;
}

.c2 img {
  height: auto;
  max-width: 100%;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-top: 1.6rem;
}

.c3 a {
  background-color: #ffd6ba;
  border-radius: 0.5rem;
  color: #555b6e;
  font-size: 1.2rem;
  font-weight: 700;
  padding: 0.8rem;
}

.c3 a:first-child {
  margin-right: 0.8rem;
}

.c0 {
  width: 100%;
  background-color: #faf9f9;
  color: #555b6e;
  border-radius: 0.5rem;
  margin: 3.2rem 0;
  padding: 2.4rem;
}

<section
  class="c0"
>
  <div
    class="c1"
  >
    <picture
      class="c2"
    >
      <img
        alt="Avatar image for the user nicomadeira"
        src="https://avatars3.githubusercontent.com/u/20728141?v=4"
      />
    </picture>
    <h2
      data-testid="username"
    >
      Nico Madeira
    </h2>
    <p>
      Front-End Developer & Mentor no projeto EducaTransforma.
    </p>
    <p>
      nicollemguimaraes@gmail.com
    </p>
    <div
      class="c3"
    >
      <a
        data-testid="followers"
        href="https://github.com/nicomadeira?tab=followers"
        rel="noopener noreferrer"
        target="_blank"
      >
        <small>
          Followers: 
          28
        </small>
      </a>
      <a
        data-testid="following"
        href="https://github.com/nicomadeira?tab=following"
        rel="noopener noreferrer"
        target="_blank"
      >
        <small>
          Following: 
          43
        </small>
      </a>
    </div>
  </div>
</section>
`
